projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss

```diff
@@ -212,7 +212,9 @@
         }
 
         @if not($date-selected-background) and $header-background {
-            $date-selected-background: $header-background;
+            @if type-of($header-background) == 'color' {
+                $date-selected-background: $header-background;
+            }
```

Comment:
This shouldn't have the type check as no color functions are called with $header-background.